src/rust/container/lib.rs

```diff
@@ -0,0 +1,384 @@
+        })
+    }
+}
+
+impl container::Server for Server {
+    fn destroy(
+        &mut self,
+        _: container::DestroyParams,
+        _: container::DestroyResults,
+    ) -> Promise<(), capnp::Error> {
+        let container_name = self.container_name.clone();
+        let docker = self.docker.clone();
+        Promise::from_future(async move {
+            let options = StopContainerOptionsBuilder::default().signal("SIGINT");
+            docker
+                .stop_container(&container_name, Some(options.build()))
+                .await
+                .map_err(|e| ContainerError::DestroyError(e.to_string()))?;
+            Ok(())
+        })
+    }
+
+    fn signal(
+        &mut self,
+        params: container::SignalParams,
+        _: container::SignalResults,
+    ) -> Promise<(), capnp::Error> {
+        let container_name = self.container_name.clone();
+        let docker = self.docker.clone();
+        Promise::from_future(async move {
+            let signo = params.get()?.get_signo();
+            let signal_name = match signo {
+                1 => "SIGHUP",
+                2 => "SIGINT",
+                3 => "SIGQUIT",
+                9 => "SIGKILL",
+                15 => "SIGTERM",
+                18 => "SIGCONT",
+                19 => "SIGSTOP",
+                20 => "SIGTSTP",
+                _ => "SIGTERM", // Default fallback
+            };
+
+            let options = StopContainerOptionsBuilder::default().signal(signal_name);
+            docker
+                .stop_container(&container_name, Some(options.build()))
+                .await
+                .map_err(|e| ContainerError::DestroyError(e.to_string()))?;
+            Ok(())
+        })
+    }
+
+    /// Starts a container.
+    /// This method doesn't handle enableInternet attribute since
+    /// it doesn't make sense for the local usage, and will be hard to
+    /// manage while still having port mapping to the host.
+    fn start(
+        &mut self,
+        raw_params: container::StartParams,
+        _: container::StartResults,
+    ) -> Promise<(), capnp::Error> {
+        let container_name = self.container_name.clone();
+        let docker = self.docker.clone();
+        Promise::from_future(async move {
+            let params = raw_params.get()?;
+            let entrypoint = params
+                .get_entrypoint()?
+                .iter()
+                .map(|s| s.unwrap().to_string().unwrap())
+                .collect::<Vec<String>>();
+            let env = params
+                .get_environment_variables()?
+                .iter()
+                .map(|s| s.unwrap().to_string().unwrap())
+                .collect::<Vec<String>>();
+            let mut create_config = CreateExecOptions::default();
+            create_config.cmd = Some(entrypoint);
+            create_config.env = Some(env);
+            let result = docker
+                .create_exec(&container_name, create_config)
```

Comment:
I believe we want this command: https://docs.rs/bollard/latest/bollard/struct.Docker.html#method.create_container

Since we wont have the container running yet

Reply:
Yes, I've updated the implementation. Thanks for the review.